drivers/pci/controller/vmd.c

```diff
@@ -644,13 +694,39 @@ static int vmd_get_bus_number_start(struct vmd_dev *vmd)
 
 		switch (BUS_RESTRICT_CFG(reg)) {
 		case 0:
-			vmd->busn_start = 0;
+			vmd->busn_start[VMD_ROOTBUS0] = 0;
 			break;
 		case 1:
-			vmd->busn_start = 128;
+			vmd->busn_start[VMD_ROOTBUS0] = 128;
 			break;
 		case 2:
-			vmd->busn_start = 224;
+			vmd->busn_start[VMD_ROOTBUS0] = 224;
+			break;
+		case 3:
+			if (features & VMD_FEAT_HAS_PCH_ROOTBUS) {
+				/* IOC start bus */
+				vmd->busn_start[VMD_ROOTBUS0] = 224;
+				/* PCH start bus */
+				vmd->busn_start[VMD_ROOTBUS1] = 225;
+			} else {
+				pci_err(dev,
+					"Unknown Bus Offset Setting (%d)\n",
+					BUS_RESTRICT_CFG(reg));
+				return -ENODEV;
+			}
+			break;
+		case 3:
+			if (features & VMD_FEAT_HAS_PCH_ROOTBUS) {
+				/* IOC start bus */
+				vmd->busn_start = 224;
+				/* PCH start bus */
+				vmd->busn_start_pch = 225;
+			} else {
+				pci_err(dev,
+					"Unknown Bus Offset Setting (%d)\n",
+					BUS_RESTRICT_CFG(reg));
+				return -ENODEV;
```

Comment:
Please consider other error codes (in each case now -ENODEV is returned from various functions)

Reply:
Fixed.